File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/ESurfaceContainerDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object ESurfaceContainerDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd15
            AppTheme.STANDARD -> R.color.cstd15
            AppTheme.BLUE -> R.color.cbd15
            AppTheme.GREEN -> R.color.cgd15
            AppTheme.RED -> R.color.crd15
            AppTheme.NOTHING -> R.color.cnod15
        }
    }
}

File: lib/layer2_core/dark/e_surface_container_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class ESurfaceContainerDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd15;
      case 'standard': return cstd15;
      case 'blue': return cbd15;
      case 'green': return cgd15;
      case 'red': return crd15;
      case 'nothing': return cnod15;
      default: return cstd15;
    }
  }
}
